xlators/cluster/afr/src/afr-common.c

```diff
@@ -3646,7 +3646,7 @@ afr_discover_unwind(call_frame_t *frame, xlator_t *this)
     return;
 
 error:
-    AFR_STACK_UNWIND(lookup, frame, local->op_ret, local->op_errno, NULL, NULL,
```

Comment:
I don't think this is the right solution. When there's an error, the only potentially valid argument is xdata (because some times it includes additional error information). All others are considered invalid/garbage. So no xlator should depend on them in case of error.

Note that depending on what failed, it could even be impossible to pass the inode back for some xlators.

If the caller needs to track the inode even in case of error, it should have saved the inode in the request path so that it can be retrieved in the cbk path without depending on data returned by the lower xlator.

Reply:
We've tried adding a "frame->local" variable to hold inodes, but it didn't work during implementation and caused the quotad process to crash. So we used the current way of using xlator to pass inodes layer by layer.